### PR TITLE
added 'error' messages; exceptions in imgConvert are now errors (not warnings)

### PR DESCRIPTION
**Pull request by whoeverest**
See this issue: https://github.com/mwilliamson/mammoth.js/issues/86

### CODE REVIEW

File: lib/results.js [hunk 1]
```diff
@@ -44,6 +45,21 @@ function warning(message) {
     };
 }
 
+function error(exception) {
+    if (typeof exception == 'string') {
```
Comment (mwilliamson) [inline lib/results.js]: I don't think this branch is needed -- could we just assume that an error will always be passed in?

File: test/mammoth.tests.js [hunk 2]
```diff
@@ -197,7 +197,7 @@ describe('mammoth', function() {
         var buffer = fs.readFileSync(docxPath);
         return mammoth.convertToHtml({buffer: buffer}).then(function(result) {
             assert.equal(result.value, '');
-            assert.deepEqual(result.messages, [results.warning("could not find external image 'tiny-picture.png', path of input document is unknown")]);
+            assert.equal(result.messages[0].message, results.error("could not find external image 'tiny-picture.png', path of input document is unknown").message);
```
Comment (mwilliamson) [inline test/mammoth.tests.js]: This weakens the assertion in that we no longer check the severity of the message (which is really the point of the exercise), nor do we check that there's exactly one message. Also, I don't think there's a need to build the expected error, only to immediately pull the message out: you may as well just use the expected message directly.

**Comment by mwilliamson** [comment]
Thanks for this. I left a couple of comments directly on the code.

One more thought: could we have a test case specifically for this change i.e. one that passes in an image converter that always fails, and checks we get the expected error?

File: lib/results.js (hunk 1 above)
Comment (whoeverest) [inline lib/results.js]: Yes, I'll remove it.

**Pull request merged**